pages/futures/index.tsx

```diff
@@ -0,0 +1,59 @@
+import { FC } from 'react';
+import Head from 'next/head';
+import styled from 'styled-components';
+import { useTranslation } from 'react-i18next';
+import { useRecoilValue } from 'recoil';
+
+import AppLayout from 'sections/shared/Layout/AppLayout';
+
+import { PageContent, MainContent, RightSideContent, FullHeightContainer } from 'styles/common';
+
+import { DesktopOnlyView } from 'components/Media';
+
+import { isWalletConnectedState, isL2State } from 'store/wallet';
+import WalletOverview from 'sections/futures/WalletOverview';
+import Markets from 'sections/futures/Markets';
+import Hero from 'sections/futures/Hero';
+
+const Futures: FC = () => {
+	const { t } = useTranslation();
+	const isWalletConnected = useRecoilValue(isWalletConnectedState);
+	const isL2 = useRecoilValue(isL2State);
+
+	return (
+		<>
+			<Head>
+				<title>{t('futures.page-title')}</title>
+			</Head>
+			<AppLayout>
+				<PageContent>
+					{!isL2 ? (
+						<h1>{t('futures.not-available-on-l1')}</h1>
+					) : (
+						<FullHeightContainer>
+							<MainContent>
+								<Hero />
+								<Markets />
+							</MainContent>
+							<DesktopOnlyView>
+								<StyledRightSideContent>
+									<WalletOverview />
+								</StyledRightSideContent>
+							</DesktopOnlyView>
+						</FullHeightContainer>
+					)}
+				</PageContent>
+			</AppLayout>
+		</>
+	);
+};
+
+const StyledRightSideContent = styled(RightSideContent)`
+	padding-left: 32px;
+	padding-right: 32px;
+	> * + * {
```

Comment:
I'm not a big fan of these CSS selectors as they'd break if anything is added into the HTML code. A bit too generic imo

Reply:
removing it, it was copied from shorts